UI/Selenium/Utilities/DriverFactory.cs

```diff
@@ -72,36 +73,101 @@ public IWebDriver InitializeDriver(BrowserType browser)
 
         public IWebDriver InitializeSauceDriver(SauceLabsOptions sauceLabsOptions, SauceLabsConfiguration config)
         {
-            AppiumOptions options = new AppiumOptions();
-            options.DeviceName = config.DeviceName;
-            options.PlatformName = config.PlatformName;
-            options.BrowserName = config.BrowserName;
-            options.AddAdditionalAppiumOption(MobileCapabilityType.AppiumVersion, config.AppiumVersion);
-            options.AddAdditionalAppiumOption(MobileCapabilityType.Orientation, config.Orientation);
-            options.AddAdditionalAppiumOption("PlatformVersion", config.PlatformVersion);
-            options.AddAdditionalAppiumOption("name", sauceLabsOptions.Name);
-
             Dictionary<string, object> SauceOptions = new Dictionary<string, object>
-            {
-                {"username",config.SauceUsername },
-                {"accessKey",config.SauceAccessKey },
-                { "name",sauceLabsOptions.Name}
-                ,{"commandTimeout",sauceLabsOptions.CommandTimeoutInSeconds }
-                ,{"idleTimeout",sauceLabsOptions.IdleTimeoutInSeconds }
-                ,{"maxDuration",sauceLabsOptions.MaxDurationInSeconds }
-                ,{"seleniumVersion",sauceLabsOptions.SeleniumVersion }
-                ,{"timeZone",sauceLabsOptions.Timezone }
-            };
+                    {
+                        {"username",config.SauceUsername },
+                        {"accessKey",config.SauceAccessKey },
+                        { "name",sauceLabsOptions.Name}
+                        ,{"commandTimeout",sauceLabsOptions.CommandTimeoutInSeconds }
+                        ,{"idleTimeout",sauceLabsOptions.IdleTimeoutInSeconds }
+                        ,{"maxDuration",sauceLabsOptions.MaxDurationInSeconds }
+                        ,{"seleniumVersion",sauceLabsOptions.SeleniumVersion }
+                        ,{"timeZone",sauceLabsOptions.Timezone }
+                    };
+            var remoteUrl = new Uri($"http://{config.SauceUsername}:{config.SauceAccessKey}{config.SauceUrl}");
 
-            foreach (var (key, value) in SauceOptions)
+            switch (config.PlatformName)
             {
-                options.AddAdditionalCapability(key, value);
-            }
+                case "Android":
```

Comment:
code for case "Android" and case "iOS" is same. they can be clubbed together

Reply:
No better not as these could be changed at another time.